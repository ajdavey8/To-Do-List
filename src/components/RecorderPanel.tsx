import React from 'react';
import styled from 'styled-components';

const Recorder = styled.div`
  background: #ffffff;
  box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  border-radius: 3px;
  margin: 32px 16px;
  padding: 16px 8px;
`

const Heading = styled.h2`
  text-align: center;
  margin: 8px;
`

const Buttons = styled.div`
  display: grid;
  grid-auto-rows: repeat(2, 1fr);
  grid-template-columns: repeat(2, 1fr);

  @media only screen and (min-width: 768px) {
    grid-template-columns: repeat(4, 1fr);
  }
`

const Button = styled.button`
  color: #494949;
  cursor: pointer;
  text-decoration: none;
  background: #ffffff;
  padding: 16px;
  border: 2px solid #f7ab1b;
  border-radius: 3px;
  transition: all 0.4s ease 0s;
  margin: 8px;

  :hover:enabled {
    color: #ffffff;
    background: #f7ab1b;
    border-color: #f7ab1b;
    transition: all 0.4s ease 0s;
  }

  :disabled {
    color: #6c757d;
  }
`

interface RecorderPanel {
  isRecording: boolean
  startRecording: () => void
  stopRecording: () => void
  clearRecording: () => void
  playRecording: () => void
}

const RecorderPanel = ({ isRecording, startRecording, stopRecording, clearRecording, playRecording }: RecorderPanel) => (
  <Recorder>
    <Heading>Record your actions</Heading>
    <Buttons>
      <Button onClick={startRecording} disabled={isRecording}>Record</Button>
      <Button onClick={stopRecording} disabled={!isRecording}>Stop Recording</Button>
      <Button onClick={clearRecording}>Clear Recording</Button>
      <Button onClick={playRecording}>Play Recording</Button>
    </Buttons>
  </Recorder>
)

export default RecorderPanel;
