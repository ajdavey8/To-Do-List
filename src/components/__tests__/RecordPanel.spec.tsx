import React from 'react';
import { render, fireEvent } from '@testing-library/react'
import RecorderPanel from '../RecorderPanel'

const startRecording = jest.fn()
const stopRecording = jest.fn()
const clearRecording = jest.fn()
const playRecording = jest.fn()


const setup = (isRecording = false) => {
  const utils = render(
    <RecorderPanel
      isRecording={isRecording}
      startRecording={startRecording}
      stopRecording={stopRecording}
      clearRecording={clearRecording}
      playRecording={playRecording}
    />
  )

  return {
    ...utils
  }
}

describe('<RecorderPanel>', () => {
  it('should renders with the passed props', () => {
    const { asFragment } = setup()
    expect(asFragment).toMatchSnapshot();
  })

  it('calls the onClick function to start recording', () => {
    const { getByText } = setup()

    fireEvent.click(getByText('Record'))
    expect(startRecording).toHaveBeenCalled()
  })

  it('calls the onClick function to stop recording', () => {
    const { getByText } = setup(true)

    fireEvent.click(getByText('Stop Recording'))
    expect(stopRecording).toHaveBeenCalled()
  })

  it('calls the onClick function to clear recording', () => {
    const { getByText } = setup()

    fireEvent.click(getByText('Clear Recording'))
    expect(clearRecording).toHaveBeenCalled()
  })

  it('calls the onClick function to play recording', () => {
    const { getByText } = setup(true)

    fireEvent.click(getByText('Play Recording'))
    expect(playRecording).toHaveBeenCalled()
  })
})