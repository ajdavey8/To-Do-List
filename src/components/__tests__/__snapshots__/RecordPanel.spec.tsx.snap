// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecorderPanel> should renders with the passed props 1`] = `[Function]`;
